### ISSUE TITLE
Error: undefinedVercel CLI 31.0.1

### ISSUE THREAD

**Issue by wottpal** [BetaHuhn/deploy-to-vercel-action#367]
Recently, we started getting this more occasionally (sometimes it works, sometimes it doesn't).

<img width="428" alt="Screenshot 2023-07-09 at 09 57 56" src="https://github.com/BetaHuhn/deploy-to-vercel-action/assets/3930150/5c77ce2c-eb65-4d8d-879c-d6218264ac3f">

**Comment by evereq** [comment]
Same issue: https://github.com/ever-co/ever-teams/actions/runs/5558865924/jobs/10154407173#step:6:17

UPDATE: we solved it by disabling the Build Cache in Vercel, see https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache. More specifically in the Vercel environment vars we added VERCEL_FORCE_NO_BUILD_CACHE and set it to 1.

**Comment by MartinLupa** [comment]
> Same issue: https://github.com/ever-co/ever-teams/actions/runs/5558865924/jobs/10154407173#step:6:17
> 
> UPDATE: we solved it by disabling the Build Cache in Vercel, see https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache. More specifically in the Vercel environment vars we added VERCEL_FORCE_NO_BUILD_CACHE and set it to 1.

Could you confirm if this environment variable could be set via the BUILD_ENV or have you set it directly in Vercel?

**Comment by evereq** [comment]
@MartinLupa we set it directly in Vercel via UI

**Comment by MartinLupa** [comment]
This issue suddenly appeared again in our project, even after setting the mentioned environment variable. Is there any update on the possible solutions?

**Comment by wottpal** [comment]
@MartinLupa Yeah environment variable & making sure the committer's git identity email is set in his github profile should help.

**Comment by psychobolt** [comment]
I have `FORCE: true` set, this will avoid using build cache as well, however I think the main issue for me is manually running `npx vercel build` with the deploy action `PREBUILT` variable enabled. If the version of CLI is out of sync (build using latest, deploy using a older release), the undefined issue occur for next.js apps. Svelte apps seem to work ok. So I globally install vercel@latest, my workflow will use the same vercel CLI version for `vercel build` and deploy action. No more undefined issues for me.
